### PR TITLE
feat(profiling): profiling onboarding sidebar

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary

**Important:** This PR depends on: https://github.com/getsentry/sentry-docs/pull/5778

Adds the new profiling onboarding sidebar behind a `profiling-onboarding-checklist` feature flag.

Examples:
image

image

### CODE REVIEW

File: static/app/components/sidebar/taskSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  align-items: center;
+  flex-grow: 1;
+  font-size: ${p => p.theme.fontSizeMedium};
+  color: ${p => (p.status === 'waiting' ? p.theme.pink300 : p.theme.green300)};
+`;
+
+export const PulsingIndicator = styled('div')`
+  ${pulsingIndicatorStyles};
+  margin-right: ${space(1)};
+`;
```
Comment (eliashussary) [inline static/app/components/sidebar/taskSidebar.tsx:73]: These all are re-used across `performance`, `replays` and `profiling`. I've put them in this file, and intend on updating `performance` and `replay` onboarding sidebars. 

I did not want to add any more changes to this PR right now.

File: src/sentry/utils/integrationdocs.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ class Platform(TypedDict):
     integrations: list[dict[str, str]]
 
 
-BASE_URL = "https://docs.sentry.io/_platforms/{}"
+INTEGRATION_DOCS_URL = os.environ.get("INTEGRATION_DOCS_URL", "https://docs.sentry.io/_platforms/")
+BASE_URL = INTEGRATION_DOCS_URL + "{}"
```
Comment (eliashussary) [inline src/sentry/utils/integrationdocs.py:39]: this allowed me to sync up with my local `sentry-docs` instance  by calling 
```bash
INTEGRATION_DOCS_URL=http://127.0.0.1:8080/_platforms/ make build-platform-assets
```

File: build-utils/integration-docs-fetch-plugin.ts [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
 import https from 'https';
 import path from 'path';
+import url from 'url';
 
 import webpack from 'webpack';
 
-const PLATFORMS_URL = 'https://docs.sentry.io/_platforms/_index.json';
+const INTEGRATIONS_DOC_URL =
+  process.env.INTEGRATION_DOCS_URL || 'https://docs.sentry.io/_platforms/';
+const PLATFORMS_URL = INTEGRATIONS_DOC_URL + '_index.json';
```
Comment (eliashussary) [inline build-utils/integration-docs-fetch-plugin.ts:14]: allows you to stay in sync with a local version of `sentry-docs` as long as we set `INTEGRATIONS_DOC_URL`

```bash
export INTEGRATION_DOCS_URL=http://127.0.0.1:8080/_platforms/
#... start dev server
```

File: build-utils/integration-docs-fetch-plugin.ts [hunk 4]
```diff
@@ -68,8 +72,13 @@ class IntegrationDocsFetchPlugin {
...
     callback
-  ) =>
-    https
+  ) => {
+    let httpClient = https;
+    if (url.parse(PLATFORMS_URL).protocol === 'http:') {
+      // @ts-ignore
+      httpClient = http;
+    }
+    return httpClient
```
Comment (JonasBa) [inline build-utils/integration-docs-fetch-plugin.ts:81]: Is this so that it works on localhost?

Comment (JonasBa) [reply build-utils/integration-docs-fetch-plugin.ts:81]: Nit, but I would prefer returns here

```
    if (url.parse(PLATFORMS_URL).protocol === 'http:') {
       return http
    }
    return https
```

Comment (eliashussary) [reply build-utils/integration-docs-fetch-plugin.ts:81]: yea, and frankly i found this fetch-plugin didn't have an effect for me, but i've aligned it to what i've done on the python side just to make sure its covered.

**Review by JonasBa — APPROVED** [review]

**Pull request merged**